src/plot.js

```diff
@@ -358,12 +369,13 @@ function applyScaleTransforms(channels, options) {
 function applyScaleTransform(channel, options) {
   const {scale} = channel;
   if (scale == null) return;
-  const {
-    type,
-    percent,
-    interval,
-    transform = percent ? (x) => x * 100 : maybeInterval(interval, type)?.floor
-  } = options[scale] ?? {};
+  let {type, percent, interval, transform} = options[scale] ?? {};
+  if (transform === undefined) {
+    if (percent) return (channel.value = channel.value.map((x) => x * 100)), undefined;
+    const i = maybeInterval(interval, type);
+    if (isTimeInterval(i)) return (channel.value = channel.value.map((d) => i.floor(coerceDate(d)))), undefined;
+    transform = i?.floor;
+  }
```

Comment:
I think it would be cleaner to refactor this into a helper like so:

```js
function maybeIntervalTransform(interval, type) {
  interval = maybeInterval(interval, type);
  return isTimeInterval(interval) ? (d) => interval.floor(coerceDate(d)) : interval?.floor;  
}
```

And then replace the default like so:

```js
transform = percent ? (x) => x * 100 : maybeIntervalTransform(interval, type)
```

In other words, I think it’s preferable to consolidate the code paths so that the interval is still promoted to a transform. And also we’d want to use `map` here instead of `channel.value.map` because `channel.value` could be a typed array.

Reply:
Also do we want to coerce to numbers for number intervals? Should we further assume that intervals only apply to numbers and dates and not other types? 🤔